velox/dwio/parquet/tests/duckdb_reader/ParquetTableScanTest.cpp

```diff
@@ -227,6 +252,51 @@ TEST_F(ParquetTableScanTest, countStar) {
   assertQuery(plan, {split}, "SELECT 20");
 }
 
+class NativeParquetTableScanTest
+    : public ParquetTableScanTest,
+      public testing::WithParamInterface<ParquetReaderType> {
+ public:
+  NativeParquetTableScanTest() : ParquetTableScanTest(GetParam()) {}
+};
+
+TEST_P(NativeParquetTableScanTest, decimalSubfieldFilter) {
+  // decimal.parquet holds two columns (a: DECIMAL(5, 2), b: DECIMAL(20, 5)) and
+  // 20 rows (10 rows per group). Data is in plain uncompressed format:
+  //   a: [100.01 .. 100.20]
+  //   b: [100000000000000.00001 .. 100000000000000.00020]
+  std::vector<int64_t> unscaledShortValues(20);
+  std::iota(unscaledShortValues.begin(), unscaledShortValues.end(), 10001);
+  loadData(
+      getExampleFilePath("decimal.parquet"),
+      ROW({"a"}, {DECIMAL(5, 2)}),
+      makeRowVector(
+          {"a"},
+          {
+              makeShortDecimalFlatVector(unscaledShortValues, DECIMAL(5, 2)),
+          }));
+
+  assertSelectWithFilter(
```

Comment:
Can we add a filter like a < 1000.7 which is also Decimal(5,2) ? Also try some values that are not DECIMAL(5,2) to test error paths of this code.

Reply:
This filter is not possible since function lt expects the decimal values to be of the same precision and scale. Column a has DECIMAL(5, 2) type but 1000.7 is DECIMAL(5, 1). I have added tests for this error case, thanks for the suggestion.